### PR TITLE
[FIX] HTTP Stream Collector better error handling

### LINKED ISSUE #1435: http stream collector: retry on regular connection problems?
http streams have a different error behavior than typical http connections. They can't timeout easily, but get interrupted which requires a restart. That happens often, but in case the error is mitigated easily, we do not need administrator's attention - an error.

This is a list of some errors happening in the last days for one of our bots
```
      2 requests.exceptions.ConnectionError: HTTPSConnectionPool(host='[...]', port=443): Read timed out.
      2 requests.exceptions.ProxyError: HTTPSConnectionPool(host='[...]', port=443): Max retries exceeded with url: [...] (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out',)))
      5 requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(505 bytes read, 7 more expected)', IncompleteRead(505 bytes read, 7 more expected))
      6 requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(503 bytes read, 9 more expected)', IncompleteRead(503 bytes read, 9 more expected))
      9 requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='[...]', port=443): Read timed out. (read timeout=30)
     19 requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(504 bytes read, 8 more expected)', IncompleteRead(504 bytes read, 8 more expected))
     68 requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))
```

Maybe some of them can be treated like timeouts in urllib3's [Retry](https://github.com/urllib3/urllib3/blob/728d9244665ef5b03103cb74d7b409ebe4f23b43/src/urllib3/util/retry.py#L29) class to be used in the [HTTPAdapter](https://2.python-requests.org/en/master/api/#requests.adapters.HTTPAdapter)

### PR DESCRIPTION
**Pull request by waldbauer-certat**
HTTP Stream collector should not raise a exception. Now we're able to define a threshold via `error_max_retries` as parameter.

**NOTE** This is a very "hacky" solution. Currently investigating the cause of this errors.

Closes #1435

### CODE REVIEW

**Comment by wagner-certat** [comment]
I deployed it on our server to check if that reduces the number of log messages.

**Comment by wagner-certat** [comment]
Looks good all in all :)

Now the logs say this:

```
2021-02-19 06:18:48,653 - some-stream-collector - INFO - Got exception ChunkedEncodingError(ProtocolError('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))), retrying (consecutive error count 1 <= 3).
...
2021-02-19 06:39:27,692 - some-stream-collector - INFO - Got exception ConnectionError(ReadTimeoutError("HTTPSConnectionPool(host='stream.shodan.io', port=443): Read timed out.")), retrying (consecutive error count 1 <= 3).
...
2021-02-19 09:59:22,054 - some-stream-collector - INFO - Got exception ChunkedEncodingError(ProtocolError('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))), retrying (consecutive error count 1 <= 3).
2021-02-19 09:59:29,918 - some-stream-collector - INFO - Got exception APIError('Unable to contact the Shodan Streaming API'), retrying (consecutive error count 2 <= 3).
...
2021-02-19 10:50:19,473 - some-stream-collector - INFO - Got exception ChunkedEncodingError(ProtocolError('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))), retrying (consecutive error count 1 <= 3).
```

Any remarks on what we want to log and on the phrasing?

File: intelmq/bots/collectors/http/collector_http_stream.py [hunk 1]
```diff
@@ -35,6 +38,8 @@ def init(self):
         self.set_request_parameters()
         self.session = create_request_session(self)
 
+        self.error_count = 0
```
Comment (schacht-certat) [inline intelmq/bots/collectors/http/collector_http_stream.py:41]: we could use a private `__error_count`, but that should not be a blocker

**Review by schacht-certat — APPROVED** [review]
LGTM

File: intelmq/bots/collectors/http/collector_http_stream.py (hunk 1 above)
Comment (wagner-certat) [reply intelmq/bots/collectors/http/collector_http_stream.py:41]: Good idea

**Review by wagner-certat — APPROVED** [review]

**Pull request merged**